webapp/src/Controller/API/JudgehostController.php

```diff
@@ -368,30 +368,41 @@ public function updateJudgingAction(Request $request, string $hostname, int $jud
             $output_compile = base64_decode($request->request->get('output_compile'));
             if ($request->request->getBoolean('compile_success')) {
                 if ($judging->getOutputCompile() === null) {
-                    $judging
-                        ->setOutputCompile($output_compile)
-                        ->setJudgehost($judgehost);
+                    $judging->setOutputCompile($output_compile);
                     $this->em->flush();
 
                     $this->eventLogService->log('judging', $judging->getJudgingid(),
                         EventLogService::ACTION_CREATE, $judging->getContest()->getCid());
                 } elseif ($judging->getResult() === Judging::RESULT_COMPILER_ERROR) {
                     // The new result contradicts a former one, that's not good.
-                    // Since the other judgehost was not successful, but we were , assume that the other judgehost
-                    // is broken and disable it.
-                    $disabled = [
-                        'kind' => 'judgehost',
-                        'hostname' => $judging->getJudgehost()->getHostname(),
-                    ];
-                    $error = new InternalError();
-                    $error
-                        ->setJudging($judging)
-                        ->setContest($judging->getContest())
-                        ->setDescription('Compilation results are different for j' . $judging->getJudgingid())
-                        ->setJudgehostlog(base64_encode('New compilation output: ' . $output_compile))
-                        ->setTime(Utils::now())
-                        ->setDisabled($disabled);
-                    $this->em->persist($error);
+                    // Since the other judgehosts were not successful, but we were , assume that the other judgehosts
+                    // are broken and disable it.
+                    $disableHostnames = [];
+                    /** @var JudgingRun $run */
+                    foreach ($judging->getRuns() as $run) {
```

Comment:
Unfortunately I think this won't happen here since if compilation failed we don't set the judging run, or do we?

Reply:
Don't we? I will test but clearly we create judgetasks and aren't judgetasks linked to runs?

Reply:
You're right

Reply:
I will still see if I can test this somehow

Reply:
Tested this by having 2 judgehosts, having 1 force a compilation failure and sleeping before reporting this, then while this happens starting a second one that also sleeps but until a bit longer and be compilation success. It seems to work.